src/Persisters/Flysystem.php

```diff
@@ -0,0 +1,124 @@
+use Rubix\ML\Persisters\Serializers\Serializer;
+use RuntimeException;
+use Stringable;
+
+/**
+ * Flysystem
+ *
+ * The flysystem persister serializes models to a file at a user-specified path, using a
+ * user-provided flysystem instance.
+ *
+ * Flysystem is a filesystem abstraction providing a unified interface for many different
+ * filesystems. (Local, Amazon S3, Azure Blob Storage, Google Cloud Storage, Dropbox etc...)
+ *
+ * @see https://flysystem.thephpleague.com
+ *
+ * @category    Machine Learning
+ * @package     Rubix/ML
+ * @author      Andrew DalPino
+ */
+class Flysystem implements Persister, Stringable
+{
+    /**
+     * The path to the model file on the filesystem.
+     *
+     * @var string
+     */
+    protected $path;
+
+    /**
+     * The filesystem implementation providing access to your backend storage.
+     *
+     * @var \League\Flysystem\FilesystemInterface
+     */
+    protected $filesystem;
+
+    /**
+     * The serializer used to convert to and from serial format.
+     *
+     * @var \Rubix\ML\Persisters\Serializers\Serializer
+     */
+    protected $serializer;
+
+    /**
+     * @param string $path
+     * @param FilesystemInterface $filesystem
+     * @param \Rubix\ML\Persisters\Serializers\Serializer|null $serializer
+     */
+    public function __construct(string $path, FilesystemInterface $filesystem, ?Serializer $serializer = null)
+    {
+        $this->path = $path;
+        $this->filesystem = $filesystem;
+        $this->serializer = $serializer ?? new Native();
+    }
+
+    /**
+     * Save the persistable model.
+     *
+     * @param \Rubix\ML\Persistable $persistable
+     * @throws \RuntimeException
+     */
+    public function save(Persistable $persistable) : void
+    {
+        $content = $this->serializer->serialize($persistable);
+        $success = $this->filesystem->put($this->path, $content->data());
+
+        if (!$success) {
+            throw new RuntimeException('Could not write to filesystem'
+                . " (Path: '{$this->path}', Filesystem: '{$this->getFlysystemAdapterName()}')");
+        }
+    }
+
+    /**
+     * Load the last model that was saved.
+     *
+     * @throws \RuntimeException
+     * @return \Rubix\ML\Persistable
+     */
+    public function load() : Persistable
+    {
+        if (!$this->filesystem->has($this->path)) {
+            throw new RuntimeException('File does not exist in filesystem'
+                . " (Path: '{$this->path}', Filesystem: '{$this->getFlysystemAdapterName()}')");
+        }
+
+        $data = new Encoding($this->filesystem->read($this->path) ?: '');
+
+        if ($data->bytes() === 0) {
+            throw new RuntimeException('File does not contain any data'
+                . " (Path: '{$this->path}', Filesystem: '{$this->getFlysystemAdapterName()}')");
+        }
+
+        return $this->serializer->unserialize($data);
+    }
+
+    protected function getFlysystemAdapterName() : string
```

Comment:
Will this method only ever be used to get a partial string representation of the object? If so maybe consider inlining this code in the __toString() method?

Reply:
Yeah. I like this idea.